ludwig/models/llm.py

```diff
@@ -418,23 +489,117 @@ def get_args(self):
             self._random_seed,
         )
 
+    def _set_pad_token(self):
+        """Sets the pad token for the tokenizer if it is not already set."""
+        # HACK(Arnav): gpt, gpt2 and llama tokenizers had no pad tokens.
+        # These recommend using eos tokens instead
+        # https://github.com/huggingface/transformers/issues/2648#issuecomment-616177044
+        # https://github.com/huggingface/transformers/issues/2630#issuecomment-1290809338
+        if any(
+            isinstance(self.tokenizer, t)
+            for t in [GPT2Tokenizer, GPT2TokenizerFast, LlamaTokenizer, LlamaTokenizerFast]
```

Comment:
Oh I see– this is because we use the raw tokenizer at this point instead of the `HFTokenizer` wrapper.

Can we move this logic into a tokenizer util function that both the HFTokenizer and this function call to prevent duplication?